File: Java/Data Structures & Algorithms Module - Java/Week10/BSTAttempt3/BSTApp.java
/*
BSTApp.java
Billy Cussen
29/11/2020
*/
public class BSTApp<T>{

    public static void main(String[]args){

        BST<Integer> myBST = new BST<Integer>();

        System.out.println("Empty: "+myBST.isEmpty());

        try{
            System.out.println("Size: "+myBST.size());
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Max: "+myBST.max());
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Min: "+myBST.min());
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Contains: "+myBST.contains(2000));
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Depth: "+myBST.depth());
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Find Node: "+myBST.findNode(2));
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Find Parent: "+myBST.findParent(3));
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            myBST.inOrder();
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        myBST.insert(15);
        myBST.insert(10);
        myBST.insert(5);
        myBST.insert(7);
        myBST.insert(2);
        myBST.insert(1);
        myBST.insert(20);
        myBST.insert(17);
        myBST.insert(22);


        System.out.println("Empty: "+myBST.isEmpty());

        try{
            System.out.println("Size: "+myBST.size());
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Max: "+myBST.max());
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Min: "+myBST.min());
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Contains 2000: "+myBST.contains(2000));
            System.out.println("Contains 22: "+myBST.contains(22));
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Depth: "+myBST.depth());
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Find Node: "+myBST.findNode(2));
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Find Parent: "+myBST.findParent(15));
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            System.out.println("Find Parent 17: "+myBST.findParent(17));
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        try{
            myBST.inOrder();
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

    }

}

File: Python/Data Structures & Algorithms Module - Python/Revision/BST1/BSTApp.py
"""
BSTApp.py
Billy Cussen
26/01/2021
"""

from BST import BST


myBST = BST()

myBST.inOrder()
print("Size: "+str(myBST.size()))
print("Depth: "+str(myBST.depth()))
print("Max: "+str(myBST.max()))
print("Min: "+str(myBST.min()))
print("Is Empty: "+str(myBST.isEmpty()))
print("Contains 5: "+str(myBST.contains(5)))
print("Contains 100: "+str(myBST.contains(100)))
print("Contains 12: "+str(myBST.contains(12)))
print("Contains 30: "+str(myBST.contains(30)))

myBST.insert(20)
myBST.insert(15)
myBST.insert(12)
myBST.insert(17)
myBST.insert(10)
myBST.insert(3)
myBST.insert(30)
myBST.insert(25)
myBST.insert(35)
myBST.insert(27)

myBST.inOrder()
print("Size: "+str(myBST.size()))
print("Depth: "+str(myBST.depth()))
print("Max: "+str(myBST.max()))
print("Min: "+str(myBST.min()))
print("Is Empty: "+str(myBST.isEmpty()))
print("Contains 5: "+str(myBST.contains(5)))
print("Contains 100: "+str(myBST.contains(100)))
print("Contains 12: "+str(myBST.contains(12)))
print("Contains 30: "+str(myBST.contains(30)))


print("Get Node (30): "+str(myBST.getNode(30).toString()))
print("Get Parent (30): "+str(myBST.getParentNode(30).toString()))
print("Get Node (20): "+str(myBST.getNode(30).toString()))
print("Get Parent (20): "+str(myBST.getParentNode(20)))
print("Get Node (27): "+str(myBST.getNode(27).toString()))
print("Get Parent (27): "+str(myBST.getParentNode(27).toString()))